### PR TITLE
Generic inherited default constructor

### LINKED ISSUE #8166: Strange behavior on coercing `typeof T` to generic constructor interface
Can anyone shed a bit of light on the following behavior

```ts
interface Constructor<T> {
	new(...args): T;
	prototype: T;
}

class A<T> { a: T; }
class B<T> extends A<T> { b: T; }
var c:Constructor<B<boolean>> = B; // error here

class A1 { a: boolean; }
class B1 extends A1 { b: boolean; }
var c1:Constructor<B1> = B1; // no error here
```
The error is:
```
Type 'typeof B' is not assignable to type 'Constructor<B<boolean>>'.
  Type 'A<any>' is not assignable to type 'B<boolean>'.
    Property 'b' is missing in type 'A<any>'.
```
I suspect that it is a correct behavior in the current design of the type system, I just don't understand two things:

1. How the error is inferred
2. Can my sample be fixed somehow without killing actual types with `<any>`

@JsonFreeman, @ahejlsberg?

### PR DESCRIPTION
**Pull request by sandersn**
Fixes #8166 by deleting an erroneous code path in `getErasedSignature`. @ahejlsberg @JsonFreeman

### CODE REVIEW

File: tests/cases/compiler/genericInheritedDefaultConstructors.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+interface Constructor<T> {
+    new(...args: any[]): T;
+    prototype: T;
+}
+
+class A<U> { a: U; }
+class B<V> extends A<V> { b: V; }
+var c:Constructor<B<boolean>> = B; // error here
```
Comment (ahejlsberg) [inline tests/cases/compiler/genericInheritedDefaultConstructors.ts]: Change comment to "shouldn't error here".

File: tests/cases/compiler/genericInheritedDefaultConstructors.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+interface Constructor<T> {
```
Comment (ahejlsberg) [inline tests/cases/compiler/genericInheritedDefaultConstructors.ts]: Maybe add this comment at the top of the file:

```typescript
// Repro from #8166
```

**Comment by ahejlsberg** [comment]
:+1: with the suggested changes.

**Comment by JsonFreeman** [comment]
Nice!

**Pull request merged**